dali/python/nvidia/dali/plugin/base_iterator.py

```diff
@@ -465,6 +467,9 @@ def checkpoints(self):
         if not self._ever_consumed:
             return self._initial_checkpoints
         else:
+            iterator_data = self._save_state()
+            for p in self._pipes:
+                p._iterator_data = iterator_data
             return [p.checkpoint() for p in self._pipes]
```

Comment:
Following the comment on readonly `p._iterator_data`

```suggestion
            iterator_data = self._save_state()
            return [p._get_checkpoint(iterator_data=iterator_data) for p in self._pipes]
```